quill-async-postgres/src/main/scala/io/getquill/context/async/ArrayAsyncEncoding.scala

```diff
@@ -0,0 +1,71 @@
+package io.getquill.context.async
+
+import java.time.LocalDate
+import java.util.Date
+
+import io.getquill.PostgresAsyncContext
+import io.getquill.context.sql.dsl.ArrayEncoding
+import io.getquill.util.Messages.fail
+import org.joda.time.{ LocalDate => JodaLocalDate, LocalDateTime => JodaLocalDateTime }
+
+import scala.reflect.ClassTag
+
+/**
+ * Base trait for encoding sql arrays via async driver.
+ * We say `array` only in scope of sql driver. In Quill we represent them as instances of Traversable
+ */
+trait ArrayAsyncEncoding extends ArrayEncoding {
```

Comment:
Wdyt about having separate traits for encoders and decoders? I'd be more similar to the rest of the encoding classes.